pytorch_lightning/core/lightning.py

```diff
@@ -1703,10 +1703,10 @@ def to_onnx(self, file_path: str, input_sample: Optional[Tensor] = None, **kwarg
             True
         """
 
-        if isinstance(input_sample, Tensor):
-            input_data = input_sample
+        if input_sample is not None:
```

Comment:
why do you need to change the if?

Reply:
the type hints already check if the type is a Tensor. This was a redundant check hence removed it.

Reply:
yes, but type check does it only in IDE not in runtime so you can pass there anything...

Reply:
Oops. didn't know that. Have added it back :]